### PR TITLE
Implements the support to context management in the lib

### PR DESCRIPTION
**Pull request by raulnegreiros**
- Adds support to context management
- Removes unused code

### CODE REVIEW

File: lib/CommandHandler.js [hunk 1]
```diff
@@ -10,17 +12,27 @@ function checkField(target, field, message) {
...
   }
 
   init() {
     this.handleResults.bind(this);
 
-    this.sock = zmq.socket('rep');
+    this.commandSocket = zmq.socket('rep');
 
-    this.sock.on("message", (request) => {
+    this.commandSocket.on("message", (request) => {
```
Comment (giovannicuriel) [inline lib/CommandHandler.js]: Just a side note: this commandSocket is missing from constructor.

File: lib/ContextHandler.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        this.client = contextClient;
+    }
+
+    saveContext(contextId, contextContext) {
+        return new Promise ((resolve, reject) => {
+            this.client.saveContext(contextId, contextContext).then(
+              () => { resolve(); },
+              (error) => { reject(error); }
+            );
+          });
```
Comment (giovannicuriel) [inline lib/ContextHandler.js]: Couldn't you just return the promise from saveContext here?
```javascript
return this.client.saveContext(contextId, contextContext);
```

BTW, I'd change ```contextContext``` to ```context```.

File: lib/ContextManagerClient.js [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+  constructor(contextManagerHost, contextManagerPort, responseTimeout) {
+    this.contextMap = {};
+    this.contextResponseTimeout = responseTimeout; // time in ms
+    this.contextSocket = null;
+    this.contextManagerHost = contextManagerHost;
+    this.contextManagerPort = contextManagerPort;
+  }
+
+  init() {
+    this.contextSocket = zmq.socket('dealer');
```
Comment (giovannicuriel) [inline lib/ContextManagerClient.js]: Wouldn't be safer to add a guard value here so the user would not mess everything up by calling init() again?

File: lib/CommandHandler.js [hunk 4]
```diff
@@ -10,17 +12,27 @@ function checkField(target, field, message) {
 }
 
 class DojotHandler {
-  constructor(dataHandler) {
+  constructor(dataHandler, contextResponseTimeout) {
+
+    let ctxResponseTimeout = contextResponseTimeout || 10000;
+    //todo the following attributes need to be exported outside?
```
[the remark this answers is not in the document]
Comment (giovannicuriel) [reply lib/CommandHandler.js]: And port.

I'm separating these variables as "X_ADDRESS" for service address (or DNS name) and "X_PORT" for port, so there's no confusion whether HOST includes or not port.

File: lib/ContextHandler.js (hunk 2 above)
Comment (giovannicuriel) [reply lib/ContextHandler.js]: This also happens in other functions in this file.

File: lib/ContextManagerClient.js (hunk 3 above)
Comment (raulnegreiros) [reply lib/ContextManagerClient.js]: The user must not use this class, but I will do it on CommandHandler.init

**Comment by giovannicuriel** [comment]
This is connected to dojot/dojot#585

**Pull request merged**